pkg/controllers/state/statenode.go

```diff
@@ -400,3 +402,52 @@ func nominationWindow(ctx context.Context) time.Duration {
 	}
 	return nominationPeriod
 }
+
+// RequireNoScheduleTaint will add/remove the karpenter.sh/disruption:NoSchedule taint from the candidates.
+// This is used to enforce no taints at the beginning of disruption, and
+// to add/remove taints while executing a disruption action.
+// nolint:gocyclo
+func RequireNoScheduleTaint(ctx context.Context, kubeClient client.Client, addTaint bool, nodes ...*StateNode) error {
```

Comment:
Musing: I'm fine with this here but wondering if if this is a state node helper or should if it should be a node helper using the node name or node key?

Reply:
Kept it here, since a statenode keeps track of the nodeclaim and node for us here. One could argue doing the same by checking nodeclaims for nodes, but I think this is an easier way to make sure we're only operating on nodes that Karpenter knows it's managing through the node <-> nodeclaim mapping.

Reply:
Not a blocker, but I think this method is questionable for a couple of reasons.

1. Static helpers -- surely there's an area of the codebase that should encapsulate this behavior. If you're passing stateful clients around, it's usually a sign that you're missing an encapsulation layer. This style of code ends up being quick stringy and difficult to make assumptions about without understanding the full picture.
2. Interfaces that accept booleans that cause divergent behavior are difficult to reason about as a caller, since you have to understand the underlying method signature. As far as I can tell, you only ever call this with `false`.